service/matching/tasklist/task_writer.go

```diff
@@ -285,6 +285,9 @@ func (w *taskWriter) sendWriteResponse(reqs []*writeTaskRequest,
 			persistenceResponse: persistenceResponse,
 		}
 
-		req.responseCh <- resp
+		select {
+		case <-w.stopCh:
+		case req.responseCh <- resp:
```

Comment:
`appendTask()` function listens `stopCh` and terminates early without reading from responseCh which was making this function stuck here. So added the select block to listen `stopCh` here as well.

Reply:
I'm not sure that's safe. the send write response on shutdown will bail, but then the outer calling function will loop again and may have side-effects. What about returning an error and making it bail out / break the writeloop on shutdown?

Reply:
I guess the writeloop is also listening to the shutdown channel, but *could* nondeterministically go either path if there's data in the write channel I think?

Reply:
> What about returning an error and making it bail out / break the writeloop on shutdown?
This is the idea and write loop will break and terminate after this function returns. The problem with existing code is that listener of `responseCh` has already given up. 

> I guess the writeloop is also listening to the shutdown channel, but could nondeterministically go either path if there's data in the write channel I think?
Yes but that's less of a concern. I think we should still fix it in separate PR. This change fixes an obvious goroutine leak which gets stuck forever

Reply:
Maybe we can make the channel a buffered channel?

Reply:
I think the effect is the same. Making it buffered might have other consequences (writer loop going faster than needed)